File: leetcode/leetcode-java-solutions/ValidPalindrome.java
package com.company;

import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

public class ValidPalindrome {
    public String s;
    public ValidPalindrome(String s) {
        this.s = s;
    }
    public boolean IsValidPalindrome(String s) {
        List <String> splitArr = Arrays.asList(s.split(""));
        List filtered = splitArr.stream().filter(character -> character.matches("[a-zA-Z0-9]")).collect(Collectors.toList());
        String joined = String.join("", filtered);
        System.out.println(joined);
        String reversedString = "";
        for (int i = filtered.size()-1; i>=0; i--) {
            reversedString += filtered.get(i);
        }
        System.out.println(reversedString);
        if (reversedString.toLowerCase().equals(joined.toLowerCase())) {
            System.out.println("true");
            return true;
        } else {
            System.out.println("false");
            return false;
        }
    }



    }

File: leetcode/leetcode-js-solutions/valid-palindrome/valid-palindrome.js
const isPalindrome = (s) => {
  const splitArr = s.split("");
  const filteredArr = splitArr.filter(character => character.match(/[A-Za-z0-9]/));
  const joinedString = filteredArr.join("");
  let reversedString = "";
  for (let i =filteredArr.length-1; i>=0; i--) {
      reversedString += filteredArr[i];
  }
  console.log(reversedString);
  console.log(joinedString)
  return reversedString.toLowerCase() === joinedString.toLowerCase() ? true : false; 
};

console.log(isPalindrome("A man, a plan, a canal: Panama"));